### ISSUE TITLE
Can a game send traces to API acting on behalf of a student?

### ISSUE THREAD

**Issue by scottbw** [e-ucm/rage-analytics#66]
Can I (as a game developer) send traces to the analytics API _on behalf of_ a student, without that student being logged in themselves? 

For example, can I send a trace for "student1" rather than "anonymous-rampant-ocelot" (etc).

On a similar note, does an actor have to exist in the Rage backend as a user to have associated traces?

I'm looking at a scenario here where I implement LTI myself, and use the RAGE backend but present my own analytics UI for the teacher view.  I can see its doable using anonymous users, but it would be much better to show real students.

Is it a valid workaround to link the anonymous animal id with the "real" student id and translate them in my app?

(NB, this isn't my final or ideal implementation scenario, I'm testing out several here to check out pros/cons)

**Comment by RotaruDan** [comment]
Yes it can be done, though signup and login is required, either by:

1 Signup as the 'username' and yourdesidedpassword (multi-signup: http://e-ucm.github.io/a2/#api-Signup-Signup, you can create your own password, or use the same as username). You have to implement this API call maybe from the client or when you connect the Analytics with the LTI server)
2 Tracker.T.login('username', yourdesiredpassword) https://github.com/e-ucm/csharp-tracker/blob/fcfbc4e8eac62e81e5492439f6f18f573719f412/TrackerAsset/TrackerAsset.cs#L687
3 Tracker.T.... - > send traces normally, they will be sent as that user

You could avoid the step 1 and 2 if you change the TrackerAsset.cs code: https://github.com/e-ucm/csharp-tracker/blob/fcfbc4e8eac62e81e5492439f6f18f573719f412/TrackerAsset/TrackerAsset.cs

Basically you would need to change "Actor.name" field from the anonymous value provided by the server, to your desired username: https://github.com/e-ucm/csharp-tracker/blob/fcfbc4e8eac62e81e5492439f6f18f573719f412/TrackerAsset/TrackerAsset.cs#L73 
This must be done after the Actor object has been received from the server, which happens exactly when the Start() method is invoked (right after Awake())

**Comment by scottbw** [comment]
Aha, I think I get it. So when a student accesses my game via LTI Launch, I create a student signup directly via the API with their LIS username, then log in to RAGE as that student, and send the traces as them. 

Makes sense. However, how will they be associated with a course and activity for the analytics?
